frontend/packages/kubevirt-plugin/src/components/modals/cd-rom-modal/cd-rom-modal.tsx

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react';
+import { Form } from '@patternfly/react-core';
+import * as _ from 'lodash';
+import {
+  FirehoseResult,
+  HandlePromiseProps,
+  withHandlePromise,
+} from '@console/internal/components/utils';
+import { getName } from '@console/shared';
+import {
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalComponentProps,
+} from '@console/internal/components/factory';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { Button } from 'patternfly-react';
+import { K8sResourceKind, k8sPatch } from '@console/internal/module/k8s';
+import { getCDsPatch } from '../../../k8s/patches/vm/vm-cdrom-patches';
+import { getRemoveDiskPatches } from '../../../k8s/patches/vm/vm-disk-patches';
+import { getVMLikeModel } from '../../../selectors/selectors';
+import {
+  getCDRoms,
+  getContainerImageByDisk,
+  getURLSourceByDisk,
+  getPVCSourceByDisk,
+} from '../../../selectors/vm/selectors';
+import { getDisks } from '../../../selectors/vm';
+import { getLoadedData } from '../../../utils';
+import { VMKind } from '../../../types';
+import { CDRomRow } from './cd-rom-row';
+import { initialDisk } from './constants';
+import './_cd-rom-modal.scss';
+
+export const VMCdRomModal = withHandlePromise((props: VMCdRomModalProps) => {
```

Comment:
```suggestion
export const VMCdRomModal = withHandlePromise(({
  vm,
  handlePromise,
  inProgress,
  errorMessage,
  persistentVolumeClaims,
  cancel,
  close,
}: VMCdRomModalProps) => {
```

and there is no need to deconstruct later